components/tidb_query_expr/src/impl_encryption.rs

```diff
@@ -60,6 +64,72 @@ pub fn sha2(
     }
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn compress(input: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    // compress implements the `COMPRESS` built-in function.
+    // MySQL doc: https://dev.mysql.com/doc/refman/5.6/en/encryption-functions.html#function_compress
+    // according to MySQL doc: Empty strings are stored as empty strings.
+    if input.is_empty() {
+        return Ok(writer.write_ref(Some(b"")));
+    }
+    let mut e = ZlibEncoder::new(input, Compression::default());
+    // preferred capacity is input length plus four bytes length header and one extra end "."
+    // max capacity is isize::max_value(), or will panic with "capacity overflow"
+    let mut vec = Vec::with_capacity((input.len() + 5).min(isize::max_value() as usize));
+    vec.resize(4, 0);
+    LittleEndian::write_u32(&mut vec, input.len() as u32);
+    match e.read_to_end(&mut vec) {
+        Ok(_) => {
+            // according to MySQL doc: append "." if ends with space
+            if vec[vec.len() - 1] == 32 {
```

Comment:
`if vec.last().unwrap() == b' '` could be more readable.

Reply:
done